picard/album.py

```diff
@@ -411,7 +411,13 @@ def _finalize_loading_track(self, track_node, metadata, artists, extra_metadata=
     def _load_tracks(self):
         artists = set()
         all_media = []
-        absolutetracknumber = 0
+
+        def _load_track(node, mm, artists, extra_metadata):
+            _load_track.absolutetracknumber += 1
+            extra_metadata['~absolutetracknumber'] = _load_track.absolutetracknumber
+            self._finalize_loading_track(node, mm, artists, extra_metadata)
+
+        _load_track.absolutetracknumber = 0
```

Comment:
why not keep this as `absolutetracknumber` variable before defining `_load_track`?

Reply:
I think it would require to use nonlocal keyword, I'll give it a try

Reply:
I did the change in [ea1fd44](https://github.com/metabrainz/picard/pull/2115/commits/ea1fd44fbee782794ae4ccccc78277c536aa66a6). Not sure which one is better, I tend to prefer the function attribute approach because it clearly links the counter to this function but the `nonlocal` approach is perhaps easier to understand.

What do you think?

Reply:
The inline definition of the function forms a closure which includes the variables defined in the scope of the `_load_tracks`  method, so this is fine.

I personally prefer how it is now. I don't really see this variable as belonging to that helper function. Rather the opposite, that variable exists in the outer context to be used there. I definitely find it easier to understand this way. But no strong opinion, it's maybe just because I'm so used to the usage of anonymous functions from other languages as well.

Reply:
Agreed, I rebased and only kept the nonlocal approach